tests/unit/test_braille/test_routing.py

```diff
@@ -147,3 +147,58 @@ def test_moveCaret_always_instantActivate(self):
 		self.assertGreaterEqual(self.cm.lastActivateTime, curTime)
 		caret = self.cm.makeTextInfo(textInfos.POSITION_CARET)
 		self.assertEquals(caret, review)
+
+
+class TestTextInfoRegionRouting(unittest.TestCase):
+	"""A test for TextInfoRegion.getTextInfoForBraillePos, which is used in braille cursor routing.
+	This test ensures that braille routes to the expected character when dealing with emoji
+	or other composites.
+	These glyphs are threated as one character by uniscribe, however they span multiple characters
+	on a braille display.
+	Note that due to the nature of this test, it relies on uniscribe to be available.
+	"""
+
+	def test_routeToEmoji(self):
+		testText = "⚠️test"
+		obj = BasicTextProvider(text=testText)
+		ti = obj.makeTextInfo(textInfos.POSITION_CARET)
+		ti.expand(textInfos.UNIT_CHARACTER)
+		self.assertEqual(ti.text, testText[:2])
+		ti.collapse(end=True)
+		ti.expand(textInfos.UNIT_CHARACTER)
+		self.assertEqual(ti.text, testText[2])
+		region = braille.TextInfoRegion(obj)
+		region.update()
+		index = 3  # Position of e
+		pos = region.rawToBraillePos[index]
+		region.routeTo(pos)
+		ti = obj.makeTextInfo(textInfos.POSITION_CARET)
+		ti.expand(textInfos.UNIT_CHARACTER)
+		self.assertEqual(ti.text, testText[index])
+
+	def test_routeToComposite(self):
+		testText = "רבְּר"
+		obj = BasicTextProvider(text=testText)
+		ti = obj.makeTextInfo(textInfos.POSITION_CARET)
+		ti.expand(textInfos.UNIT_CHARACTER)
+		self.assertEqual(ti.text, testText[0])
+		ti.collapse(end=True)
+		ti.expand(textInfos.UNIT_CHARACTER)
+		self.assertEqual(ti.text, testText[1:4])
+		ti.collapse(end=True)
+		ti.expand(textInfos.UNIT_CHARACTER)
+		self.assertEqual(ti.text, testText[4])
+		region = braille.TextInfoRegion(obj)
+		region.update()
+		index = 1  # Position of ב
+		pos = region.rawToBraillePos[index]
+		region.routeTo(pos)
+		ti = obj.makeTextInfo(textInfos.POSITION_CARET)
+		ti.expand(textInfos.UNIT_CHARACTER)
+		self.assertEqual(ti.text, testText[1:4])
+		index = 3  # Position of ּ
+		pos = region.rawToBraillePos[index]
+		region.routeTo(pos)
+		ti = obj.makeTextInfo(textInfos.POSITION_CARET)
+		ti.expand(textInfos.UNIT_CHARACTER)
+		self.assertEqual(ti.text, testText[1:4])
```

Comment:
@coderabbitai Do you have concrete suggestions on how to cover edge cases?

Reply:
This is lovely. Please make this a commitable suggestion.

Reply:
I can't commit these. Please wrap them in a suggestion block so it will be actionable

Reply:
I'm afraid this was still not what I intended, since this removed code
I fixed it in commit 7bc812653ca6fa20e363631efd6f779d4791a1d1 , please learn from it.